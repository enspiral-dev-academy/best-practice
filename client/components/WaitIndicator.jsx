import React from "react";

import { wrappedWith, pendingContext } from "../wrappers"

const style = {
  width: "40px",
  height: "40px",
};

export function WaitIndicator({ pending }) {
  return pending ? <img style={style} src="/animated-circle.gif" /> : null;
}

export default wrappedWith(pendingContext)(WaitIndicator)
