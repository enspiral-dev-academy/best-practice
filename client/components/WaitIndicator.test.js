import React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";

import { WaitIndicator } from "./WaitIndicator";

test("renders loading indicator gif if pending is true", () => {
  const { asFragment } = render(<WaitIndicator pending={true} />);
  expect(asFragment()).toMatchSnapshot();
});

test("renders null if pending is false", () => {
  const { asFragment } = render(<WaitIndicator pending={false} />);
  expect(asFragment()).toMatchSnapshot();
});
