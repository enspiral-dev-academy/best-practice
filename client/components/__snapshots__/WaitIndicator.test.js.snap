// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loading indicator gif if pending is true 1`] = `
<DocumentFragment>
  <img
    src="/animated-circle.gif"
    style="width: 40px; height: 40px;"
  />
</DocumentFragment>
`;

exports[`renders null if pending is false 1`] = `<DocumentFragment />`;
